pkg/authenticator/k8s/authenticator.go

```diff
@@ -48,9 +48,27 @@ const (
 	nameTypeIP    = 7
 )
 
+// Init the authenticator struct
+func (auth *Authenticator) Init(config *common.ConfigurationInterface) (common.AuthenticatorInterface, error) {
+	var cfg = (*config).(*Config)
+	authn, err := New(*cfg)
+	if err != nil {
+		return nil, fmt.Errorf(log.CAKC019)
+	}
+
+	return authn, nil
+}
+
+func (auth *Authenticator) InitWithAccessToken(config *common.ConfigurationInterface, token access_token.AccessToken) (common.AuthenticatorInterface, error) {
+	log.Debug(log.CAKC058)
+	var cfg = (*config).(*Config)
+
+	return NewWithAccessToken(*cfg, token)
+}
+
 // New creates a new authenticator instance from a token file
-func New(config authnConfig.Config) (*Authenticator, error) {
-	accessToken, err := file.NewAccessToken(config.TokenFilePath)
+func New(config Config) (*Authenticator, error) {
```

Comment:
I think it would be good for us to update this to `NewAuthenticator` to make it explicit what this method is creating.

Reply:
Done